### PR TITLE
feature: add support for CodeDeployToECS deployments

### PR DESCRIPTION
**Pull request by inhumantsar**
## what
* Adds support for CodeDeployToECS deployments, mainly to support Blue/Green deploys

## why
* Blue/Green deployments in ECS are a commonly used feature that is well-supported by CodeDeploy and CodePipeline.
* These changes allow for direct integration with [cloudposse/code_deploy_blue_green](https://github.com/cloudposse/terraform-aws-code-deploy)
* I want to do Blue/Green deploys without completely re-implementing this module myself 😉

## references
* [Similar implementation](https://github.com/gnokoheat/ecs-with-codepipeline-example-by-terraform/blob/master/code-pipeline.tf#L188-L208)

### CODE REVIEW

**Comment by inhumantsar** [comment]
The resources that failed in this test aren't from anything I changed right?

```
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: Error: error waiting for ECS Cluster (arn:aws:ecs:us-east-2:126450723953:cluster/eg-test-ecs-codepipeline) to become Available: unexpected state 'INACTIVE', wanted target 'ACTIVE'. last error: %!s(<nil>)
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121:   on main.tf line 26, in resource "aws_ecs_cluster" "default":
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121:   26: resource "aws_ecs_cluster" "default" {
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: Error: Error creating IAM Role eg-test-ecs-codepipeline-task: EntityAlreadyExists: Role with name eg-test-ecs-codepipeline-task already exists.
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 	status code: 409, request id: 3b11ddea-32a0-4c7e-bac7-81ea13532d42
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121:   on .terraform/modules/ecs_alb_service_task/main.tf line 112, in resource "aws_iam_role" "ecs_task":
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121:  112: resource "aws_iam_role" "ecs_task" {
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: Error: Error creating IAM Role eg-test-ecs-codepipeline-exec: EntityAlreadyExists: Role with name eg-test-ecs-codepipeline-exec already exists.
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 	status code: 409, request id: 4cc51ecf-5feb-45b3-b1b9-7a4fa57a1c34
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121:   on .terraform/modules/ecs_alb_service_task/main.tf line 190, in resource "aws_iam_role" "ecs_exec":
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121:  190: resource "aws_iam_role" "ecs_exec" {
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121: 
TestExamplesComplete 2021-08-02T11:21:58Z command.go:121:
```

Am I able to retest?

/test all

**Comment by nitrocode** [comment]
It's possible those resources came from another PR.

File: main.tf [hunk 1]
```diff
@@ -206,13 +206,58 @@ data "aws_caller_identity" "default" {
...
+  taskdef_writer = <<EOF
+      - echo Writing task definition file...
+      - echo "${var.codedeploy_taskdefinition}" > ${var.codedeploy_taskdefinition_path}
+  EOF
+  appspec_writer = <<EOF
+      - echo Writing app spec file...
+      - echo "${var.codedeploy_appspec}" > ${var.codedeploy_appspec_path}
+  EOF
+
+  buildspec = var.buildspec != "" ? var.buildspec : <<EOF
```
Comment (nitrocode) [inline main.tf:222]: Instead of a heredoc, a separate `.tmpl` or `.tpl` file and the use of the `templatefile` function would be better.

**Comment by inhumantsar** [comment]
i'm going to add an example to this in the next few days once i've had a chance to better test things.

also: i opted against doing the standard codepipeline ecr trigger. it works fine this way and imho the standard way would make this module a little bulky but curious what you think.

**Pull request closed without merging**